src/go/rpk/pkg/cli/acl/acl.go

```diff
@@ -72,17 +72,35 @@ group.
 
 PRINCIPALS
 
-All ACLs require a principal. A principal is composed of two parts: the type
-and the name. Within Redpanda, only one type is supported, "User". The reason
-for the prefix is that a potential future authorizer may add support for
-authorizing by Group or anything else.
+All ACLs require a principal (or a role; see ROLES, below). A principal is
+composed of two parts: the type and the name. Within Redpanda, only one type
+is supported, "User". The reason for the prefix is that a potential future
+authorizer may add support for authorizing by Group or anything else.
 
 When you create a user, you need to add ACLs for it before it can be used. You
 can create / delete / list ACLs for that user with either "User:bar" or "bar"
 in the --allow-principal and --deny-principal flags. This command will add the
 "User:" prefix for you if it is missing. The wildcard '*' matches any user.
 Creating an ACL with user '*' grants or denies the permission for all users.
 
+ROLES
+
+ACLs can also be bound to a role. A role has only one part: the name. In
+contrast to principals in general, there is no need to supply type. If a
+type-like prefix is present, it will be considered part of the role name
+and processed accordingly.
+
+For example '--allow-role "foo"' specifies a role "foo", whereas
+'--allow-role "User:foo"' specifies a role "User:foo". In either case, the
+result is assigned to an internal namespace for roles and will not collide
+with similarly named user principals.
+
+When you create a role, you need to add ACLs for it before it can be used.
+You can create / delete / list ACLs for that role with "<name>" in the
+--allow-role and --deny-role flags. Note that the wildcard '*' is not
+allowed here. Creating an ACL bound to role '*' has no effect, but it will
+not result in an error.
```

Comment:
the doc is perfect, but why doesn't it yield an error, since it has no effect? 

If the procedure has no effect, outputting an error might help the user to not make that mistake.

Reply:
Ah, I should have tried it by hand. `rpk` _does_ report an error, but the CLI process still gives a zero return code, which is why I couldn't get it to fail in my integration tests. Looks like this:

```bash
$ rpk acl create --allow-principal "RedpandaRole:*" --operation all --topic all
PRINCIPAL       HOST  RESOURCE-TYPE  RESOURCE-NAME  RESOURCE-PATTERN-TYPE  OPERATION  PERMISSION  ERROR
RedpandaRole:*  *     TOPIC          all            LITERAL                ALL        ALLOW       INVALID_REQUEST
```

This is definitely by design, but I'll update the doc to reflect that.